lib/twitter_streaming_monitor.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+class TwitterStreamingMonitor
+  POLLING_INTERVAL = 60
+
+  @monitor = nil
+
+  attr_reader :twitter_ids
+  attr_accessor :tweet_consumer, :monitor_thread, :exit_flag
+
+  def initialize(twitter_ids)
+    @twitter_ids = twitter_ids
+    @exit_flag = false
+    @tweet_consumer = nil
+    @monitor_thread = nil
+  end
+
+  class << self
+    attr_accessor :monitor
+  end
+
+  def self.run
+    return if monitor&.alive?
+
+    twitter_ids = SynchronizedObjectHolder.new { TwitterProfile.active_twitter_ids }
+    self.monitor = TwitterStreamingMonitor.new(twitter_ids)
+    monitor.run
+  end
+
+  def self.stop
+    monitor.stop
+  end
```

Comment:
per our discussion, refactor to remove redundant class methods

Reply:
Removed.